hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/DataSourceOptions.scala

```diff
@@ -411,6 +411,11 @@ object DataSourceWriteOptions {
     .markAdvanced()
     .withDocumentation("Key generator class, that implements `org.apache.hudi.keygen.KeyGenerator`")
 
+  val KEYGENERATOR_TYPE: ConfigProperty[String] = ConfigProperty
+    .key("hoodie.datasource.write.keygenerator.type")
+    .defaultValue(KeyGeneratorType.SIMPLE.name)
+    .withDocumentation("Key generator type.")
```

Comment:
we might need to add more docs here. for eg, which one takes precedence among the key gen class and key gen type. thats why we removed it sometime back. 
whats the real need to add it back ?

Reply:
This is to support bootstrap keygen type. I can remove it if it's not necessary.

Reply:
yes.

Reply:
Sg. Fixed.